pandas/core/indexes/period.py

```diff
@@ -704,36 +705,38 @@ def get_loc(self, key, method=None, tolerance=None):
         -------
         loc : int
         """
+        if is_integer(key):
+            ordinal = key
+        else:
+            key = self._cast_to_period_object(key)
+            ordinal = iNaT if key is NaT else key.ordinal
+
         try:
-            return self._engine.get_loc(key)
+            if tolerance is not None:
+                tolerance = self._convert_tolerance(tolerance, np.asarray(key))
+            return self._int64index.get_loc(ordinal, method, tolerance)
         except KeyError:
-            if is_integer(key):
-                raise
+            raise KeyError(key)
 
+    def _cast_to_period_object(self, key):
+        if is_period_arraylike(key):
```

Comment:
any particular reason for this instead of is_period_dtype?  We use this much less frequently

Reply:
@jbrockmendel 
Because conficts with error message in some test cases. So i change error message.